test/unit/utils_test.js

```diff
@@ -68,6 +68,1210 @@ describe('Utils', function() {
+        serverKP,
+        clientKP.publicKey(),
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      clock.tick(200);
+
+      const transaction = new StellarSdk.Transaction(
+        challenge,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      expect(
+        StellarSdk.Utils.readChallengeTx(
+          challenge,
+          serverKP.publicKey(),
+          StellarSdk.Networks.TESTNET,
+        ),
+      ).to.eql({
+        tx: transaction,
+        clientAccountID: clientKP.publicKey(),
+      });
+    });
+
+    it("throws an error if transaction sequenceNumber is different to zero", function() {
+      let keypair = StellarSdk.Keypair.random();
+
+      const account = new StellarSdk.Account(keypair.publicKey(), "100");
+      const transaction = new StellarSdk.TransactionBuilder(
+        account,
+        txBuilderOpts,
+      )
+        .setTimeout(30)
+        .build();
+
+      let challenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(() =>
+        StellarSdk.Utils.readChallengeTx(
+          challenge,
+          keypair.publicKey(),
+          StellarSdk.Networks.TESTNET,
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        /The transaction sequence number should be zero/,
+      );
+    });
+
+    it("throws an error if transaction source account is different to server account id", function() {
+      let keypair = StellarSdk.Keypair.random();
+
+      const challenge = StellarSdk.Utils.buildChallengeTx(
+        keypair,
+        "GBDIT5GUJ7R5BXO3GJHFXJ6AZ5UQK6MNOIDMPQUSMXLIHTUNR2Q5CFNF",
+        "SDF",
+        300,
+        StellarSdk.Networks.TESTNET,
+      );
+
+      let serverAccountId = StellarSdk.Keypair.random().publicKey();
+
+      expect(() =>
+        StellarSdk.Utils.readChallengeTx(
+          challenge,
+          serverAccountId,
+          StellarSdk.Networks.TESTNET,
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        /The transaction source account is not equal to the server's account/,
+      );
+    });
+
+    it("throws an error if transaction doestn't contain exactly one operation", function() {
+      let keypair = StellarSdk.Keypair.random();
+      const account = new StellarSdk.Account(keypair.publicKey(), "-1");
+      const transaction = new StellarSdk.TransactionBuilder(
+        account,
+        txBuilderOpts,
+      )
+        .setTimeout(30)
+        .build();
+
+      transaction.sign(keypair);
+      const challenge = transaction
+        .toEnvelope()
+        .toXDR("base64")
+        .toString();
+
+      expect(() =>
+        StellarSdk.Utils.readChallengeTx(
+          challenge,
+          keypair.publicKey(),
+          StellarSdk.Networks.TESTNET,
+        ),
+      ).to.throw(
+        StellarSdk.InvalidSep10ChallengeError,
+        /The transaction should contain exactly one operation/,
+      );
+    });
```

Comment:
Could you add another test similar to this test that has multiple operations, that shows that if a challenge transaction has more than one operation it also fails? This test is testing what happens if no operation is present, but would be good to test what happens if there are multiple and not exactly one.

Reply:
Agreed! I'll create that extra test